### ISSUE TITLE
~50% drop between test and valid performance [SSv2 - Uniformer S16]

### ISSUE THREAD

**Issue by namhar** [Sense-X/UniFormer#42]
Hello,

I'm observing drop in performance between test and validation using Uniformer S16 on SSv2

1. The label mapping provided in somesomev2_rgb_test_split.txt lists label 0 for all instances.
   1. Obtained test labels from another repo and matched to ensure indices match the train and valid label mapping
   2. Training from scratch (including on k400) yields validation results of 67.99 top 1 acc. (slightly higher but comparable to paper & github reported results)
   3. When evaluate this model on test. Results drop to 31.86 top 1 acc (similar drop for top 5 acc)
   4. Looking through the provided logs [here](https://drive.google.com/file/d/12Hc63B0e_AqeB4Av3dW0hCQq-CstpNy1/view), don't see the test set being invoked. I see interleaved train and validation at 1x1 (clips x crops) until the end where 1x3 evaluation for validation is done.
   6. I want to confirm the reported values in the paper and github are on test and not validation.
   
2. The pretrained [checkpoint](https://drive.google.com/file/d/1-Wq22024HB7tT8kPX6eSMn2GUNMI-_gY/view?usp=sharing) (trained on ssv2 + k400) shows very poor performance on the label mapping provided across test/train/valid for ssv2. ie. the model checkpoints seem to be trained on a different label mapping. Do you happen to have these available?

I appreciate help tracking these down.

Thanks

**Comment by Andy1621** [comment]
Thanks for your good questions!

1. Actually, `somesomev2_rgb_test_split.txt` is never used, because we do not have the truth. For SSV2, the popular methods only test on `somesomev2_rgb_validation_split.txt`. As shown in [MViT](https://github.com/facebookresearch/SlowFast/blob/52fb753f8f703b306896afc5613978db0c3c6695/slowfast/datasets/ssv2.py#L87-L93), they also adopt the validation dataset.
<img width="473" alt="image" src="https://user-images.githubusercontent.com/24236723/166017606-94334551-7f99-432e-8595-2a6910110c3e.png">

2. I have checked the data_list, the data is exactly what I used. Maybe the gap is from the different rates? In our experiments, we extract the frames in [30FPS](https://github.com/Sense-X/UniFormer/blob/main/video_classification/DATASET.md). Some papers extract the frames with the original frame rates. You can check whether the frame numbers is same.

**Comment by namhar** [comment]
Hi @Andy1621, thanks for the quick response! 

1. Got it, thanks for clarifying.
2. Will investigate this, I used the steps in the uniformer repo to process the data but will take a closer look.

**Comment by Andy1621** [comment]
As there is no more activity, I am closing the issue, don't hesitate to reopen it if necessary.
